airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/stream_slicer.py

```diff
@@ -28,3 +28,45 @@ def stream_slices(self) -> Iterable[StreamSlice]:
 
         :return: List of stream slices
         """
+
+    def set_parent_state(self, stream_state: StreamState) -> None:
+        """
+        Set the state of the parent streams.
+
+        This method should only be defined if the slicer is based on some parent stream and needs to read this stream
+        incrementally using the state.
+
+        Args:
+            stream_state (StreamState): The state of the streams to be set. The expected format is a dictionary that includes
+                                        'parent_state' which is a dictionary of parent state names to their corresponding state.
+                                        Example:
+                                        {
+                                            "parent_state": {
+                                                "parent_stream_name_1": { ... },
+                                                "parent_stream_name_2": { ... },
+                                                ...
+                                            }
+                                        }
+        """
+        pass
+
+    def get_parent_state(self) -> Optional[Mapping[str, StreamState]]:
```

Comment:
let's leave the method as abstract to force the concrete class to implement it (or intentionally implement a no-op)

Reply:
Fixed